pkg/dataconn/wire.go

```diff
@@ -6,39 +6,46 @@ import (
 	"fmt"
 	"io"
 	"net"
+	"runtime"
 )
 
 type Wire struct {
-	conn   net.Conn
-	writer *bufio.Writer
-	reader io.Reader
+	conn      net.Conn
+	writer    *bufio.Writer
+	reader    io.Reader
+	byteOrder binary.ByteOrder
 }
 
 func NewWire(conn net.Conn) *Wire {
+	var byteOrder binary.ByteOrder = binary.LittleEndian
+	if runtime.GOARCH == "s390x" {
+		byteOrder = binary.BigEndian
```

Comment:
Could you explain more about this change?

cc @keithalucas @derekbit

Reply:
s390 is different from other architectures. 

ref: https://community.ibm.com/community/user/ibmz-and-linuxone/blogs/javier-perez1/2021/01/19/endianness-guidance-for-open-source-projects

Reply:
Yes, exactly s390x is a Big Endian architecture unlike x86_64 which is Little Endian.